From 0c9b8b166de52f8cc23dc63aade2c15490cda733 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Mon, 18 Aug 2014 19:56:04 -0400
Subject: [scsi] qla2xxx: Add changes to obtain ISPFX00 adapters product information in accordance with firmware update

Message-id: <1408391817-6389-17-git-send-email-cdupuis@redhat.com>
Patchwork-id: 87917
O-Subject: [PATCH RHEL 7.1 e-stor 16/69] qla2xxx: Add changes to obtain ISPFX00 adapters product information in accordance with firmware update.
Bugzilla: 1089346
RH-Acked-by: Rob Evers <revers@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Armen Baloyan <armen.baloyan@qlogic.com>

scsi commit id 03eb912a4d559c4c565814eaf2e18866c3d54487

Signed-off-by: Armen Baloyan <armen.baloyan@qlogic.com>
Signed-off-by: Saurav Kashyap <saurav.kashyap@qlogic.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/qla2xxx/qla_attr.c b/drivers/scsi/qla2xxx/qla_attr.c
index dd0b72f..d435e76 100644
--- a/drivers/scsi/qla2xxx/qla_attr.c
+++ b/drivers/scsi/qla2xxx/qla_attr.c
@@ -956,10 +956,6 @@ qla2x00_model_name_show(struct device *dev, struct device_attribute *attr,
 {
  scsi_qla_host_t *vha = shost_priv(class_to_shost(dev));
 
- if (IS_QLAFX00(vha->hw))
-  return snprintf(buf, PAGE_SIZE, "%s\n",
-      vha->hw->mr.product_name);
-
  return snprintf(buf, PAGE_SIZE, "%s\n", vha->hw->model_number);
 }
 
diff --git a/drivers/scsi/qla2xxx/qla_dbg.c b/drivers/scsi/qla2xxx/qla_dbg.c
index 9cd767c..f6103f5 100644
--- a/drivers/scsi/qla2xxx/qla_dbg.c
+++ b/drivers/scsi/qla2xxx/qla_dbg.c
@@ -12,6 +12,7 @@
  * |             Level            |   Last Value Used  |     Holes |
  * ----------------------------------------------------------------------
  * | Module Init and Probe        |       0x015b       | 0x4b,0xba,0xfa |
+ * |                              |                    | 0x0x015a |
  * | Mailbox commands             |       0x1187       | 0x111a-0x111b  |
  * |                              |                    | 0x1155-0x1158  |
  * |                              |                    | 0x1018-0x1019  |
diff --git a/drivers/scsi/qla2xxx/qla_mr.c b/drivers/scsi/qla2xxx/qla_mr.c
index 5c44dd0..fec48bd 100644
--- a/drivers/scsi/qla2xxx/qla_mr.c
+++ b/drivers/scsi/qla2xxx/qla_mr.c
@@ -1967,8 +1967,8 @@ qlafx00_fx_disc(scsi_qla_host_t *vha, fc_port_t *fcport, uint16_t fx_type)
  if (fx_type == FXDISC_GET_CONFIG_INFO) {
   struct config_info_data *pinfo =
       (struct config_info_data *) fdisc->u.fxiocb.rsp_addr;
-  memcpy(&vha->hw->mr.product_name, pinfo->product_name,
-      sizeof(vha->hw->mr.product_name));
+  strcpy(vha->hw->model_number, pinfo->model_num);
+  strcpy(vha->hw->model_desc, pinfo->model_description);
   memcpy(&vha->hw->mr.symbolic_name, pinfo->symbolic_name,
       sizeof(vha->hw->mr.symbolic_name));
   memcpy(&vha->hw->mr.serial_num, pinfo->serial_num,
diff --git a/drivers/scsi/qla2xxx/qla_mr.h b/drivers/scsi/qla2xxx/qla_mr.h
index c8b8acd..1416e5e 100644
--- a/drivers/scsi/qla2xxx/qla_mr.h
+++ b/drivers/scsi/qla2xxx/qla_mr.h
@@ -304,7 +304,9 @@ struct register_host_info {
 #define QLAFX00_TGT_NODE_LIST_SIZE (sizeof(uint32_t) * 32)
 
 struct config_info_data {
- uint8_t  product_name[256];
+ uint8_t  model_num[16];
+ uint8_t  model_description[80];
+ uint8_t  reserved0[160];
  uint8_t  symbolic_name[64];
  uint8_t  serial_num[32];
  uint8_t  hw_version[16];
@@ -491,7 +493,6 @@ struct qla_mt_iocb_rsp_fx00 {
 #define FX00_DEF_RATOV 10
 
 struct mr_data_fx00 {
- uint8_t product_name[256];
  uint8_t symbolic_name[64];
  uint8_t serial_num[32];
  uint8_t hw_version[16];
diff --git a/drivers/scsi/qla2xxx/qla_os.c b/drivers/scsi/qla2xxx/qla_os.c
index 5f505d1..10bcf42 100644
--- a/drivers/scsi/qla2xxx/qla_os.c
+++ b/drivers/scsi/qla2xxx/qla_os.c
@@ -2852,12 +2852,8 @@ skip_dpc:
 
  qla2x00_dfs_setup(base_vha);
 
- if (IS_QLAFX00(ha))
-  ql_log(ql_log_info, base_vha, 0x015a,
-      "QLogic %s.\n", ha->mr.product_name);
- else
-  ql_log(ql_log_info, base_vha, 0x00fb,
-      "QLogic %s - %s.\n", ha->model_number, ha->model_desc);
+ ql_log(ql_log_info, base_vha, 0x00fb,
+     "QLogic %s - %s.\n", ha->model_number, ha->model_desc);
  ql_log(ql_log_info, base_vha, 0x00fc,
      "ISP%04X: %s @ %s hdma%c host#=%ld fw=%s.\n",
      pdev->device, ha->isp_ops->pci_info_str(base_vha, pci_info),
-- 
1.7.1